doc/python_reference.rst

```diff
@@ -425,6 +425,8 @@ EEG referencing:
    :toctree: generated/
 
    calculate_head_pos_ctf
+   calculate_chpi_coil_locs
+   calculate_head_pos_chpi_coil_locs
```

Comment:
calculate_head_pos_chpi_coil_locs -> calculate_head_pos or calculate_head_pos_fif  to be consistent with calculate_head_pos_ctf ?

note that we have public function called plot_head_positions

or maybe we could have a single function calculate_head_pos that takes CTF or fif files as input?

naive question: why do you do it in 2 steps for fif and 1 step for CTF?

Reply:
> note that we have public function called plot_head_positions

We have `read_head_pos`, `write_head_pos`, and `head_pos` is the arg for `maxwell_filter` so really `plot_head_positions` is the outlier

> calculate_head_pos_chpi_coil_locs -> calculate_head_pos or calculate_head_pos_fif to be consistent with calculate_head_pos_ctf ?
>
> or maybe we could have a single function calculate_head_pos that takes CTF or fif files as input?
> 
> naive question: why do you do it in 2 steps for fif and 1 step for CTF?

cHPI head position estimation is a two-step process:

1. calculating the locations of the N HPI coils as a function of time, and then
2. using these to calculate time-varying dev_head_t

`calculate_head_pos_ctf` only does the second step, because CTF has already done the first one, namely computed the coil locations, and stored the results in their file. When you call `calculate_chpi_coil_locs` (which can be done for FIF, Artemis123, or any system that has coils continuously emitting HPI freqs -- maybe even CTF?), you are doing the process that CTF has already done.

The naming is meant to reflect this: `calculate_chpi_coil_locs` does the first step, and either `calculate_head_pos_ctf` or `calculate_head_pos_chpi_coil_locs` will compute a `head_pos` depending on whether you want to use the CTF coil location estimates or our own `chpi_coil_locs`, respectively.

As to why it should be separated into two steps instead of a single one:

1. The first step is *much* slower than the second step. It can take 5 minutes to compute coil locations for a 20 minute recording with a lot of movement.
2. Just having the coil positions themselves (output of the first step) can be useful for QA, as you can get a quick time-varying cHPI quality assessment based on the inter-coil distannces (we do this at UW). And the `coil_locs` output is just a dict of `ndarray` so it can easily be written to disk and read back with with `h5io`.
3. The first step in general you won't change the parameters. The defaults are good and reasonable, so you should only really need to run it once per recording.
4. The second step is much faster (5-10 sec for a 20 minute recording) but in general is the one where you might need to iteratively / interactively tweak params to get good results -- `adjust_dig`, `gof_limit`, and `dist_limit` all matter, especially for child recordings where they move a lot.

Reply:
I think it's better if we have same the functions that deal internally with the fact that it's ctf or fif, eventually with optional coil locs if already present in the file. So

mne.chpi.compute_chpi_coil_locs or maybe mne.chpi.compute_chpi_locs ?
mne.chpi.comput_head_pos that can take ctf or fif and requires chpi_locs as parameters if fif is passed.

and then head_pos can be passed to maxwell_filter.

note that to be consistent it should be called compute and not calculate.

wdyt?

Reply:
I don't like the distinction for FIF vs CTF. It's not about FIF/Elekta because the coil locs can be for FIF/Elekta, Artemis123, or even recomputed in theory for CTF (and also new systems if they ever come out). I'd rather abandon the vendor specific naming and stay with what they actually functionally do:

- compute_chpi_locs
- compute_head_pos_chpi_locs
- compute_head_pos_ctf_chpi_locs

Or so. Or maybe make it so that there is a `extract_ctf_coil_locs` and then for CTF data you call this and then call `compute_head_pos_chpi_locs`, i.e.:

- compute_chpi_locs
- extract_chpi_locs_ctf
- compute_head_pos_chpi_locs

Reply:
Actually the last proposal seems cleanest, and this naming is even better:

- compute_chpi_locs (for any vendor)
- extract_chpi_locs_ctf (only for ctf)
- compute_head_pos(coil_locs, ...) (for any vendor)

Reply:
This last one is the API I like best, out of the ones discussed here so far.

Reply:
Looking at the code it also makes things more DRY as there was duplicated (and outdated) code in `calculate_head_pos_ctf` that now lives in a unified `compute_head_pos` function.

Reply:
Pushed a commit with this API, @agramfort feel free to see if you agree it's cleaner